File: app/src/main/java/ru/prsolution/winstrike/ui/common/BottomDecoratorHelper.java
package ru.prsolution.winstrike.ui.common;

import android.graphics.Rect;
import android.support.v7.widget.RecyclerView;
import android.view.View;

public class BottomDecoratorHelper extends RecyclerView.ItemDecoration {

    private final int bottomOffset;

    public BottomDecoratorHelper(int bottomOffset) {
        this.bottomOffset = bottomOffset;
    }

    @Override
    public void getItemOffsets(Rect outRect, View view, RecyclerView parent, RecyclerView.State state) {
        super.getItemOffsets(outRect, view, parent, state);
        int dataSize = state.getItemCount();
        int position = parent.getChildPosition(view);
        if (dataSize > 0 && position == dataSize - 1) {
            outRect.set(0, 0, 0, bottomOffset);
        } else {
            outRect.set(0, 0, 0, 0);
        }

    }
}

File: app/src/main/kotlin/ru/prsolution/winstrike/presentation/utils/custom/BottomDecoratorHelper.kt
package ru.prsolution.winstrike.presentation.utils.custom

import android.graphics.Rect
import android.view.View

import androidx.recyclerview.widget.RecyclerView

class BottomDecoratorHelper(private val bottomOffset: Int) : RecyclerView.ItemDecoration() {

    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
        super.getItemOffsets(outRect, view, parent, state)
        val dataSize = state.itemCount
        val position = parent.getChildPosition(view)
        if (dataSize > 0 && position == dataSize - 1) {
            outRect.set(0, 0, 0, bottomOffset)
        } else {
            outRect.set(0, 0, 0, 0)
        }
    }
}
